tt_metal/common/core_descriptor.cpp

```diff
@@ -8,9 +8,15 @@
 
 namespace tt {
 
-const core_descriptor_t &get_core_descriptor_config(chip_id_t device_id, const uint8_t num_hw_cqs) {
+const core_descriptor_t &get_core_descriptor_config(chip_id_t device_id, const uint8_t num_hw_cqs, CoreType dispatch_core_type) {
     // {arch : {product : {num hardware command queues : config}}}
     static std::unordered_map<ARCH, std::unordered_map<std::string, std::unordered_map<uint8_t, core_descriptor_t>>> config_by_arch;
+    // TODO: is there a better way to do this?
+    static CoreType previous_dispatch_core_type;
+    if (previous_dispatch_core_type != dispatch_core_type) {
```

Comment:
is this to allow toggling between using tensix / eth dispatch? should we build  `config_by_arch` for both tensix/eth and then pick between the two?

Reply:
Yea :/ I thought about adding the tensix/eth toggle but that would propagate to other similar static objects in this file. I didn't want to add it for now, because there doesn't seem to be any real use case to toggle tensix vs eth dispatch within same process. We could just assert here for that case instead..

Reply:
yeah maybe assert so that it isn't implicitly allowed